src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -367,9 +367,14 @@ def step(
             )
 
         # 3. Clip or threshold "predicted x_0"
+        if self.config.sample_max_value is None and self.config.clip_sample:
```

Comment:
I don't really understand this change - what are we doing here exactly? Is the problem that thresholding is done after clamping in Kandinsky?

Reply:
@patrickvonplaten 

I just need to clamp it and then thresholding 

Currently, it will only do thresholding __OR__ clamping, never both together, so if you set `clip_smple` to be `True`, it will NOT clip it if `threholding` is also `True`.  I checked the config for IF,  both`clip_sample` and `thresholding` are set to be `True` there, but only threholding will be applied https://huggingface.co/DeepFloyd/IF-I-XL-v1.0/blob/main/scheduler/scheduler_config.json

I'm not sure why we need this `if ... else... `  logic , I think it's confusing. if we only want threholding, we should just set `clip_sample` to be `False`. Removing it now will break IF I think

Reply:
The reason for why thresholding and clipping is done in an `if, else` statement is because it's more or less the same operation.

If you do thresholding your output **has to be** clipped to [-1, 1] as can be seen here: https://github.com/huggingface/diffusers/blob/59aefe9ea6d56245d257439f757384fb1ac5d477/src/diffusers/schedulers/scheduling_ddim.py#L280

So in the case of IF even if the `self.config.clip_sample` would be run afterward it would make no difference as the output is already scaled to be between -1 and 1. We could/should probably add a warning that running `thresholding` and `clipping` together doesn't make sense and should be avoided. For IF it would probably be best if we just set `clip_sample=False` or is this used for anything else ? cc @williamberman 

@yiyixuxu for Kandinsky the current "working" scheduler config is:
```
DDPMScheduler(
  clip_sample= True,
  clip_sample_range=2.0,
  sample_max_value=None,
  num_train_timesteps= 1000,
  prediction_type= "epsilon",
  variance_type= "learned_range",
  thresholding= True,
  beta_schedule= "linear",
  beta_start= 0.00085,
  beta_end=0.012
)
```

with this PR. I'm fairly certain that the following config **without** any changes to `scheduling_ddim.py` is the same:

```
DDPMScheduler(
  clip_sample= False,
  sample_max_value=2.0,
  num_train_timesteps= 1000,
  prediction_type= "epsilon",
  variance_type= "learned_range",
  thresholding= True,
  beta_schedule= "linear",
  beta_start= 0.00085,
  beta_end=0.012
)
```

Could you try this out?

Reply:
I think the current logic since clip sample is the default, we just prioritize thresholding if its also set. So we're ok with both being set but if so we just do the threshold clipping

Reply:
And yeah it's an if/else because thresholding is a form of clipping so it doesn't make sense to do both

Reply:
Guess overall it just doesn't make too much sense to set both no? Think a warning like "Your clip sample is useless and doesn't do anything" could make sense

Reply:
> Could you try this out?

will try and report back :)

Reply:
yeah no issues with either

Reply:
@patrickvonplaten the cat image looks same to me! will revert 

is math the same, or just that difference too small that we can ignore? I understood clip sample and dynamic thresholding are trying to do the same thing, but when I looked at the formula, I thought clamping first would cause the threshold to be different and cause a difference in the result, no the case it seems? 
![yiyi_test_pipe_ddpm_patrick_out](https://github.com/huggingface/diffusers/assets/12631849/4f86e254-32d3-4ef9-9dae-fd106af11aeb)

Reply:
Think math is the same here actually